contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,143 @@
+    /// allowing the owner to add and remove them and anyone to deposit any
+    /// fungible token among the available capabilities
+    ///
+    pub resource Switchboard: FungibleToken.Receiver, SwitchboardPublic {
+        
+        /// receiverCapabilities
+        /// Dictionary holding the fungible token receiver capabilities, 
+        /// indexed by the fungible token vault type
+        ///
+        access(contract) var receiverCapabilities: {Type: Capability<&{FungibleToken.Receiver}>}
+
+        /// addNewVault adds a new fungible token receiver capability
+        ///                    to the switchboard resource
+        ///            
+        /// 
+        /// Parameters: capability: The capability to expose a certain fungible
+        /// token vault deposit function through {FungibleToken.Receiver} that
+        /// will be added to the switchboard
+        ///
+        pub fun addNewVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // to store inside the switchboard
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference type as key for storing the capability
+            self.receiverCapabilities[vaultRef.getType()] = capability
+            // Emit the event that indicates that a new capability has been added
+            emit VaultCapabilityAdded(type: vaultRef.getType())
+        }
+
+        /// removeVault removes a fungible token receiver capability 
+        ///                       from the switchboard resource
+        /// 
+        /// Parameters: capability: The capability to a fungible token vault 
+        ///                         to be removed from the switchboard
+        ///
+        pub fun removeVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // store inside the switchboard            
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference to find the capability to remove
+            self.receiverCapabilities.remove(key: vaultRef.getType())
+            // Emit the event that indicates that a new capability has been removed
+            emit VaultCapabilityRemoved(type: vaultRef.getType())            
+        }
+        
+        /// deposit Takes a fungible token vault and routes it to the proper
+        ///         fungible token receiver capability for depositing it
+        /// 
+        /// Parameters: from: The deposited fungible token vault resource
+        ///        
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let depositedVaultCapability = self.receiverCapabilities[from.getType()] ?? 
```

Comment:
oh I am surprised how this key is working: `from.getType()` 

@joshuahannan you know better but aren't they are different types? One is restricted and One is Vault ?

Reply:
The reference acts as a way to access the underlying type, so any built-in functions like `getType()` would return the same thing as the underlying type would. I agree it does seem a little weird, but that is the intended behavior

Reply:
Yeah, that's the reason I asked this on stackoverflow https://stackoverflow.com/questions/72408565/reference-types-in-cadence, I believe it was you the first to answer me!!

Reply:
hey @alilloig , I remember that question, I mean it was reference vs nonReference getType comparison right? 

Actually it makes sense, but this one has got me thinking long. I think this one is wrong, but I need to ask Bastian maybe.

Reply:
In which way do you think it's wrong? @turbolent can you take a look at it?

Reply:
yeah I asked him already, all good